Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs

```diff
@@ -115,27 +115,12 @@ private Run ConvertToSarif(Weakness_Catalog cweXml, string version, string type)
             switch (type.ToLower())
             {
                 case "comprehensive":
-                    switch (version.ToLower())
-                    {
-                        case "4.3":
-                            cweTaxonomy.Name = Constants.CWE_Comprehensive_V43.Name;
-                            cweTaxonomy.Guid = Constants.CWE_Comprehensive_V43.Guid;
-                            cweTaxonomy.ReleaseDateUtc = Constants.CWE_Comprehensive_V43.ReleaseDate;
-                            cweTaxonomy.InformationUri = new Uri("https://cwe.mitre.org/data/published/cwe_v4.3.pdf");
-                            cweTaxonomy.DownloadUri = new Uri("https://cwe.mitre.org/data/xml/cwec_v4.3.xml.zip");
-                            cweTaxonomy.ShortDescription = new MultiformatMessageString { Text = "The MITRE Common Weakness Enumeration" };
-                            break;
-                        case "4.4":
-                            cweTaxonomy.Name = Constants.CWE_Comprehensive_V44.Name;
-                            cweTaxonomy.Guid = Constants.CWE_Comprehensive_V44.Guid;
-                            cweTaxonomy.ReleaseDateUtc = Constants.CWE_Comprehensive_V44.ReleaseDate;
-                            cweTaxonomy.InformationUri = new Uri("https://cwe.mitre.org/data/published/cwe_v4.4.pdf");
-                            cweTaxonomy.DownloadUri = new Uri("https://cwe.mitre.org/data/xml/cwec_v4.4.xml.zip");
-                            cweTaxonomy.ShortDescription = new MultiformatMessageString { Text = "The MITRE Common Weakness Enumeration" };
-                            break;
-                        default:
-                            break;
-                    }
+                    cweTaxonomy.Name = Constants.CWE_Comprehensive[version].Name;
+                    cweTaxonomy.Guid = Constants.CWE_Comprehensive[version].Guid;
+                    cweTaxonomy.ReleaseDateUtc = Constants.CWE_Comprehensive[version].ReleaseDate;
+                    cweTaxonomy.DownloadUri = new Uri(Constants.CWE_Comprehensive[version].DownloadUri);
+                    cweTaxonomy.InformationUri = new Uri(Constants.CWE_Comprehensive[version].InformationUri);
```

Comment:
Do we have any tests covering this? There was some renaming that happened with the constants as part of this, we need some confirmation that this functions as expected.

Reply:
the result of this repo that is referenced by users are the .sarif files,
the Tools.sln C# code in this repo is for dev use only to help generate the .sarif files, I think that is the reason there was no test code in this repo.

Let me know how you think, happy to add the Test csproj and work on tests.
Currently the Tools.sln only have one project "Tools.csproj"
I checked the naming convention in SARIF SDK, looks like it would be "Test.UnitTests.Tools.csproj" if to add.

Reply:
Yes, please add. We're at a point where we should not be making changes without test coverage.

Reply:
Thanks! Working on it now.

Reply:
added the test project and added test in the build pipeline.